src/test/scala/chiselTests/ChiselEnum.scala

```diff
@@ -401,6 +401,30 @@ class ChiselEnumSpec extends ChiselFlatSpec with Utils {
     assertTesterPasses(new CastToUIntTester)
   }
 
+  // This is a bug but fixing it may break user code
+  // See https://github.com/chipsalliance/chisel/issues/4159
```

Comment:
Mega nits, also throughout. Full sentence comments should be a period:

```suggestion
  // This is a bug, but fixing it may break user code.
  // See: https://github.com/chipsalliance/chisel/issues/4159
```